emscripten.py

```diff
@@ -2117,14 +2095,18 @@ def create_asm_end(exports):
     // EMSCRIPTEN_END_ASM
     ''' % (exports)
 
-  return '''
+  rtn = '''
 
   return %s;
 })
 // EMSCRIPTEN_END_ASM
 (asmGlobalArg, asmLibraryArg, buffer);
 ''' % (exports)
 
+  if not shared.Settings.MINIMAL_RUNTIME and shared.Settings.ASSERTIONS:
+    rtn += 'asm = wrapNativeFunctionsWithAssertions(asm)\n'
+  return rtn
```

Comment:
Is this for the exports or the imports? (perhaps the name could clarify that, see other comment)

for the imports, the asyncify code in emscripten.py is on line 2701 (`Asyncify.instrumentWasmImports`). For the exports, preamble.js 925 (`Asyncify.instrumentWasmExports`). It would be good to have those alongside each other as much as possible.